src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java

```diff
@@ -214,17 +214,17 @@ public static String getNodeNameSelectorFromNodeIdentifier(NodeInfo node) {
   }
 
   public static void uploadPipelineFromFile(String filename) {
+    String pipelineNameXPathSelector = "//div[contains(@class, 'PipelineName')]";
+    WebElement element = locateElementByLocator(By.xpath(pipelineNameXPathSelector));
     WebElement uploadFile = SeleniumDriver.getDriver()
       .findElement(By.xpath("//*[@id='pipeline-import-config-link']"));
 
     File pipelineJSONFile = new File(Constants.FIXTURES_DIR + filename);
     String filePath = pipelineJSONFile.getAbsolutePath();
     uploadFile.sendKeys(filePath);
 
-    String pipelineNameXPathSelector = "//div[contains(@class, 'PipelineName')]";
     try {
-      SeleniumDriver.getWaitDriver().until(ExpectedConditions
-          .stalenessOf(locateElementByLocator(By.xpath(pipelineNameXPathSelector))));
+      SeleniumDriver.getWaitDriver().until(ExpectedConditions.stalenessOf(element));
```

Comment:
nit: Changes in this file are for resolving flaky tests it seems. Just for tracking, could you please provide your observations arriving at this fix?

Reply:
It seems like that the uploadFile.sendKeys method call in the previous step, is resetting the pipelineName element to be fetched. Hence there could be a race condition between the UI updating the elements and the locateElementByLocator method call. And it might be possible that the locateElementByLocator does not get any webelement instance in the UI tree. Hence to avoid that I am getting the WebElement before the sendKeys call.